File: uniexetask.services/WorkShopService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using uniexetask.core.Interfaces;
using uniexetask.core.Models;
using uniexetask.services.Interfaces;
using System.Threading.Tasks;

namespace uniexetask.services
{
    public class WorkShopService : IWorkShopService
    {
        private readonly IUnitOfWork _unitOfWork;
        public WorkShopService(IUnitOfWork unitOfWork) 
        {
            _unitOfWork = unitOfWork;
        }

        public async System.Threading.Tasks.Task CreateWorkShop(Workshop workShop)
        {
            workShop.Status = "Status 1";
            await _unitOfWork.WorkShops.InsertAsync(workShop);
            _unitOfWork.Save();
            return;
        }

        public void DeleteWorkShop(int workShopId)
        {
            _unitOfWork.WorkShops.Delete(workShopId);
            _unitOfWork.Save();
            return;
        }

        public async Task<IEnumerable<Workshop>> GetWorkShops()
        {
            return await _unitOfWork.WorkShops.GetAsync();
        }

        public async System.Threading.Tasks.Task UpdateWorkShop(Workshop workShop)
        {
            var workShopToUpdate = await _unitOfWork.WorkShops.GetByIDAsync(workShop.WorkshopId);

            if (workShopToUpdate != null)
            {
                workShopToUpdate.Name = workShop.Name;
                workShopToUpdate.Description = workShop.Description;
                workShopToUpdate.StartDate = workShop.StartDate;
                workShopToUpdate.EndDate = workShop.EndDate;
                workShopToUpdate.Location = workShop.Location;
                workShopToUpdate.RegUrl = workShop.RegUrl;
                workShopToUpdate.Status = workShop.Status;

                _unitOfWork.WorkShops.Update(workShopToUpdate);
                await _unitOfWork.SaveAsync();
            }
            else
            {
                throw new Exception("Workshop not found");
            }
        }

    }
}

File: unitask.frontend/src/services/workShopService.js
import axios from "axios";
import Workshop from "../models/WorkShop";

export const getWorkShops = () => {
    return axios.get('https://localhost:7289/api/workshop')
        .then(response => {
            console.log('Workshop data:', response);
            if (response.data.success) {
                return response.data.data.map(workshop => new Workshop(
                    workshop.workshopId,
                    workshop.name,
                    workshop.description,
                    new Date(workshop.startDate),
                    new Date(workshop.endDate),
                    workshop.location,
                    workshop.regUrl,
                    workshop.status
                ));
            }
            return [];
        })
        .catch(error => {
            console.error('Error fetching workshop data:', error);
            return [];
        });
};

export async function insertWorkshop(data) {
    try {
        const response = await axios.post(`https://localhost:7289/api/workshop`, data);

        if (response.data.success) {
            return response.data.data; 
        } else {
            console.error("Error inserting workshop:", response.data.errorMessage);
            return false; 
        }
    } catch (error) {
        console.error("Error inserting workshop:", error);
        throw error; 
    }
}


export async function updateWorkshop(data) {
    try {
        const { workshop_id, ...rest } = data;
        const workshopData = {
            workshopId: workshop_id,
            ...rest,
        };
        const response = await axios.put(`https://localhost:7289/api/workshop`, workshopData);

        if (response.data.success) {
            return response.data.data; 
        } else {
            console.error("Error updating workshop:", response.data.errorMessage);
            return false; 
        }
    } catch (error) {
        console.error("Error updating workshop:", error);
        throw error; 
    }
}

export function deleteWorkshop(id) {
    return axios.delete(`https://localhost:7289/api/workshop/${id}`)
        .then(response => {
            console.log('Workshop deleted:', response);
            return response.data; 
        })
        .catch(error => {
            console.error('Error deleting workshop:', error);
            throw error; 
        });
}
